### DISCUSSION TITLE
Integrate Emgu.CV into Xamarin.Forms app

### DISCUSSION THREAD

**Discussion by milanm** [emgucv/emgucv#687] [Q&A] [answered]
I have commercial license for Emgu.CV and downloaded the whole package locally. How to integrate it into Xamarin.Forms app ?

**Comment by emgucv** [comment, marked as answer]
Depends on which platform you are targeting.

For windows, instruction can be found here:
https://www.emgu.com/wiki/index.php/Download_And_Installation#Commercial_Release

For iOS, see this:
https://www.emgu.com/wiki/index.php/Download_And_Installation#iOS

For Android, see this:
https://www.emgu.com/wiki/index.php/Download_And_Installation#Android

**Reply by milanm** [reply]
I'm using Xamarin.Forms project, with both platforms Android and iOS. For these two platforms I know, but not sure how to integrate in a standard library project in Xamarin.Forms.

**Reply by emgucv** [reply]
If you are using nuget packages, a general setup should look something like this:

MyLibrary -> Add Reference to "Emgu.CV" nuget package
MyXamarinWindowsApp -> Add Reference to "MyLibrary" and "Emgu.CV.runtime.windows" nuget package
MyXamarinMacApp -> Add Reference to "MyLibrary" and "Emgu.CV.runtime.mac" nuget package
MyXamarinAndroidApp -> Add Reference to "MyLibrary" and "Emgu.CV.runtime.android" nuget package
MyXamarinIOSApp -> Add Reference to "MyLibrary" and "Emgu.CV.runtime.ios" nuget package

**Reply by milanm** [reply]
Thanks! I downloaded commercial package. Where I can find those nuget packages ?

**Reply by emgucv** [reply]
You can follow instructions from this link to find your user name and url to download the nuget packages:
https://www.emgu.com/wiki/index.php/Download_And_Installation#Commercial_Release

**Reply by milanm** [reply]
Yes I get access to https://emgu.com/emgucv4_pro/ and what I need to download, i.e., where are those nuget packages located exactly?

**Reply by emgucv** [reply]
Latest Android nuget package can be found here:
https://emgu.com/emgucv4_pro/android/4.5.5/Emgu.CV.runtime.android.4.5.5.4823.nupkg
iOS nuget pacakge can be found here:
https://emgu.com/emgucv4_pro/ios/4.5.5/Emgu.CV.runtime.ios.4.5.5.4823.nupkg

**Reply by milanm** [reply]
O.K. Thx. I managed to install it.

**Reply by milanm** [reply]
I have one more question, I'm building and distributing my app through https://appcenter.ms/, but there is no nuget feed for these packages:

 NU1101: Unable to find package Emgu.CV.runtime.ios. No packages exist with this id in source(s): nuget.org

Do you know how to solve it ?

**Reply by emgucv** [reply]
You probably need to setup a private nuget feed, upload the Emgu CV nuget packages to the feed, then configure AppCenter to use the private Nuget Feed:
https://intercom.help/appcenter/en/articles/1335668-how-to-restore-a-private-nuget-feed

**Reply by Peppegio** [reply]
Hi! I'm also interesting in emgucv integration with Xamarin.Forms to elaborate edge detection.
Did you have any problems? Are you able to generate a size limited package (less than 40MB) for the stores?  
Thanks!
